pkg/apis/cilium/types_network.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TunnelMode string
+
+const (
+	VXLan TunnelMode = "vxlan"
+	Geneve TunnelMode = "geneve"
+	NoTunnel TunnelMode = "disabled"
+)
+
+type KubeProxyReplacementMode string
+
+const (
+	Strict  KubeProxyReplacementMode = "strict"
+	Probe   KubeProxyReplacementMode = "probe"
+	Partial KubeProxyReplacementMode = "partial"
+)
+
+type IPAMOptionMode string
+
+const (
+	Crd IPAMOptionMode = "crd"
+	Eni IPAMOptionMode = "eni"
+	None IPAMOptionMode = "default"
+)
+
+// KubeProxy configuration for cilium
+type KubeProxy struct {
+	// KubeProxy_enabled is used to define whether KubeProxy is required or not.
+	KubeProxy_enabled bool `json:"kubeproxy_enabled"`
+}
+
+// ExternalIPs configuration for cilium
+type ExternalIP struct {
+        // ExternalIP_enabled is used to define whether ExternalIP address is required or not.
+        ExternalIP_enabled bool `json:"externalip_enabled"`
+}
+
+// Debug level option for cilium
+type Debug struct {
+	// Debug_enabled is used to define whether Debug is required or not.
+	Debug_enabled bool `json:"debug_enabled"`
+}
+
+// Hubble enablement for cilium
+type Hubble struct {
+	// Hubble_enabled is used to define whether Hubble is required or not.
+	Hubble_enabled bool `json:"hubble_enabled"`
+}
+
+// Nodeport configuration for cilium
+type Nodeport struct {
+        // Nodeport_enabled  is used to define whether Nodeport is required or not.
+        Nodeport_enabled bool `json:"ipv4_enabled"`
+}
+
+// IPv4 configuration for cilium
+type IPv4 struct {
+	// Enabled is used to define whether IPv4 address is required or not.
+	IPV4_enabled bool `json:"ipv4_enabled"`
+}
+
+// IPv6 configuration for cilium
+type IPv6 struct {
+        // Enabled is used to define whether IPv4 address is required or not.
+        IPV6_enabled bool `json:"ipv6_enabled"`
+}
+
+// NetworkSpec is the configuration specific to a node
+type NetworkSpec struct {
+        // Debug configuration to be enabled or not 
+        //
+        // +optional
+	Debug Debug `json:"debug,omitempty"`
+	// KubeProxy configuration to be enabled or not 
+        //
+        // +optional
+        KubeProxy KubeProxy `json:"kubeproxy"`
```

Comment:
json tag is not needed for internal types. Please apply this suggestion to the rest of the fields in this file. 
```suggestion
        KubeProxy KubeProxy
```